crit/mempages.go

```diff
@@ -193,3 +195,96 @@ func (mr *MemoryReader) GetShmemSize() (int64, error) {
 
 	return size, nil
 }
+
+// PatternMatch represents a match when searching for a pattern in memory.
+type PatternMatch struct {
+	Vaddr   uint64
+	Length  int
+	Context int
+	Match   string
+}
+
+// SearchPattern searches for a pattern in the process memory pages.
+func (mr *MemoryReader) SearchPattern(pattern string, context, chunkSize int) ([]PatternMatch, error) {
+	if context < 0 {
+		return nil, errors.New("context size cannot be negative")
+	}
+
+	// Set a default chunk size of 10MB to be read at a time
+	if chunkSize <= 0 {
+		chunkSize = 10 * 1024 * 1024
+	}
+
+	regexPattern, err := regexp.Compile(pattern)
```

Comment:
It is very common for passwords to contain characters like `*` `|` `.` `(` `)`. However, these are regex metacharacters that users would need to manually escape when searching.

Would it make sense to use a boolean argument indicating whether to escapes metacharacters?

For example, something like the following:

```Go
if escapeRegExpCharacters {
    pattern = regexp.QuoteMeta(pattern)
}
```
with `checkpointctl memparse` providing two options, where the first option escapes metacharacters:
- `--search=<string pattern>`
- `--search-regex=<regex pattern>`

Reply:
This completely makes sense. Thanks @rst0git )